server/api/assessment/updateItem.put.ts

```diff
@@ -1,18 +1,60 @@
-import { AssessmentItem } from "../../../db/models";
+import { AssessmentItem, Boundary, Boundary_User } from "../../../db/models";
 
 export default defineEventHandler(async (event) => {
   const body = await readBody(event);
 
-  const assessmentItem = await AssessmentItem.findByPk(body.id);
+  const rawToken = getCookie(event, "tirtoken");
+  let userId: number;
+  if (rawToken) {
+    userId = decodeToken(rawToken);
+  } else {
+    throw createError({
+      statusCode: 401,
+      statusMessage: "Unknown User.",
+    });
+  }
 
-  if (assessmentItem) {
-    for (const key in body) {
-      assessmentItem.setDataValue(key, body[key]);
-    }
+  const boundary = await Boundary.findByPk(body.BoundaryId, {
+    attributes: ["id", "name", "ownerId"],
+    include: [
+      {
+        model: Boundary_User,
+      },
+    ],
+  });
+  const isOwner = boundary?.dataValues.ownerId === userId;
```

Comment:
As discussed, might want to do a research spike into the CASL stuff for not just page access but also for actions that you can do.  

https://github.com/mitre/heimdall2/blob/master/apps/backend/src/casl/casl-ability.factory.ts

https://casl.js.org/v6/en/package/casl-vue